nzshm_common/grids/region_grid.py

```diff
@@ -41,5 +49,74 @@ def load(self):
         return self.grid.load()
 
 
-def load_grid(gri_name: str) -> Tuple[float, float]:
-    return RegionGrid[gri_name].load()
+def load_grid(grid_name: str) -> List[LatLon]:
+    """
+    Load values from a region grid as `LatLon` pairs.
+
+    Examples:
+        >>> from nzshm_common import grids
+        >>> grids.load_grid("NZ_0_1_NB_1_0")
+        [
+            LatLon(latitude=-46.1, longitude=166.4),
+            LatLon(latitude=-46.0, longitude=166.4),
+            LatLon(latitude=-45.9, longitude=166.4),
+            ...
+        ]
+    """
+    return RegionGrid[grid_name].load()
+
+
+def get_location_grid(location_grid_name: str, resolution: float = DEFAULT_RESOLUTION) -> Iterable[CodedLocation]:
+    """
+    Get all coded locations within a grid.
+
+    Note:
+        When downsampling to a lower resolution, duplicate location values will be removed.
+
+    Parameters:
+        location_grid_name: a valid member key from RegionGrid (e.g. "NZ_0_1_NB_1_1")
+        resolution: the resolution of the CodedLocation values generated
+
+    Returns:
+        a list of coded locations
+
+    Examples:
+        >>> from nzshm_common import grids
+        >>> grids.get_location_grid("NZ_0_1_NB_1_0")
+        [
+            CodedLocation(lat=-35.109, lon=173.262, resolution=0.001),
+            CodedLocation(lat=-35.22, lon=173.97, resolution=0.001),
+            ...
+        ]
+    """
+    grid_values = load_grid(location_grid_name)
+    coded_at_resolution = partial(CodedLocation.from_tuple, resolution=resolution)
+    # Remove duplicate coordinates from collection, preserving order.
+    location_list = []
+    for loc in map(coded_at_resolution, grid_values):
+        if loc not in location_list:
+            location_list.append(loc)
+
+    return location_list
+
+
+def get_location_grid_names() -> Iterable[str]:
```

Comment:
Parking as above.

Reply:
Merge as-is, but we need a ticket to reconcile the various data sources (across grids, location lists, bins).

Reply:
-> #45